### DISCUSSION TITLE
incoming QPI traffic on Haswell

### DISCUSSION THREAD

**Discussion by abasak24** [opcm/pcm#140] [Q&A] [answered]
Hi,
It looks like QPIIncomingTraffic is not available on Haswell Architecture. Could you please explain why? 

bool incomingQPITrafficMetricsAvailable() const
    {
        return getQPILinksPerSocket() > 0 &&
            (
                cpu_model == PCM::NEHALEM_EX
            ||  cpu_model == PCM::WESTMERE_EX
            ||  cpu_model == PCM::JAKETOWN
            ||  cpu_model == PCM::IVYTOWN
            || (cpu_model == PCM::SKX && cpu_stepping > 1)
               );
    }

**Comment by opcm** [comment, marked as answer]
This metric  was over counting on Haswell and Broadwell therefore it has been disabled.

**Comment by abasak24** [comment]
got it, thanks for your response!

**Comment by ogbrugge** [comment]
Should we add a comment here stating the finding that the counter is overcounting? This will prevent future questions.

**Comment by ogbrugge** [comment]
I also mean this in general, whenever a cpu model is left out of a list I think it should require a comment as to why it was left out.

**Comment by soramichi** [comment]
Can you please point a document that explains this phenomenon?

I once observed this (like 16 MB of QPI IO even when nothing interesting is going on), but at that time I though this traffic was to maintain the cache coherence and I didn't know that it was due to overcountng (bug)?
